metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -2841,3 +2844,344 @@ def test_add_dataset_data_product_transformation():
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove digits
+    config: List[Union[re.Pattern, str]] = [r"\d+"]
+    expected_user_emails: List[str] = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id@example.com",
+        "email_id:id@example.com",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+
+def test_clean_owner_urn_transformation_remove_pattern(mock_datahub_graph):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove `example.*`
+    config: List[Union[re.Pattern, str]] = [r"@example\.\S*"]
+    expected_user_emails: List[str] = [
+        "ABCDEF:email_id",
+        "ABCDEF:123email_id",
+        "email_id",
+        "email_id",
+        "email_test:XYZ",
+        "email_id:id1",
+        "email_id:id2",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+
+def test_clean_owner_urn_transformation_remove_capital_letters(mock_datahub_graph):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
```

Comment:
Please add a test case for 
`"email_test:XYabZ@example.com",`

I want to see what happens in that case for this regex

Reply:
here, RegEx is expecting the entire STRING between `:` and `@` should be in CAPITAL LETTERS and only ALPHABETS; if that matches it will remove it...

`"email_test:XYabZ@example.com",` here the string between `:` and `@` is in CAMEL CASE; which will not match the RegEx and hence will not be removed